koku/api/iam/models.py

```diff
@@ -102,25 +103,36 @@ class Tenant(TenantMixin):
     auto_create_schema = False
 
     def _check_clone_func(self):
-        LOG.info(f'Verify that clone function "{self._CLONE_SCHEMA_FUNC_SIG}" exists')
+        LOG.info(f'Verify that clone function "{self._CLONE_SCHEMA_FUNC_NAME}" exists')
         res = dbfunc_exists(
-            conn, self._CLONE_SCHEMA_FUNC_SCHEMA, self._CLONE_SHEMA_FUNC_NAME, self._CLONE_SCHEMA_FUNC_SIG
+            conn, self._CLONE_SCHEMA_FUNC_SCHEMA, self._CLONE_SCHEMA_FUNC_NAME, self._CLONE_SCHEMA_FUNC_SIG
         )
         if not res:
-            LOG.warning(f'Clone function "{self._CLONE_SCHEMA_FUNC_SIG}" does not exist')
-            LOG.info(f'Creating clone function "{self._CLONE_SCHEMA_FUNC_SIG}"')
+            LOG.warning(f'Clone function "{self._CLONE_SCHEMA_FUNC_NAME}" does not exist')
+            LOG.info(f'Creating clone function "{self._CLONE_SCHEMA_FUNC_NAME}"')
             apply_sql_file(conn.schema_editor(), self._CLONE_SCHEMA_FUNC_FILENAME, literal_placeholder=True)
             res = dbfunc_exists(
-                conn, self._CLONE_SCHEMA_FUNC_SCHEMA, self._CLONE_SHEMA_FUNC_NAME, self._CLONE_SCHEMA_FUNC_SIG
+                conn, self._CLONE_SCHEMA_FUNC_SCHEMA, self._CLONE_SCHEMA_FUNC_NAME, self._CLONE_SCHEMA_FUNC_SIG
+            )
+        else:
+            LOG.info(f'Clone function "{self._CLONE_SCHEMA_FUNC_NAME}" exists. Not creating.')
+
+        LOG.info(f'Verify clone function "{self._CLONE_SCHEMA_FUNC_NAME}" is up to date')
+        if (hash := get_function_hash(conn, self._CLONE_SCHEMA_FUNC_NAME)) != self._CLONE_SCHEMA_FUNC_SHA256:
+            LOG.info(f'Updating function "{self._CLONE_SCHEMA_FUNC_NAME}"')
+            apply_sql_file(conn.schema_editor(), self._CLONE_SCHEMA_FUNC_FILENAME, literal_placeholder=True)
+            LOG.info(
+                f'Updated function "{self._CLONE_SCHEMA_FUNC_NAME}": '
+                f"{get_function_hash(conn, self._CLONE_SCHEMA_FUNC_NAME)}"
             )
         else:
-            LOG.info("Clone function exists")
+            LOG.info(f'Clone function "{self._CLONE_SCHEMA_FUNC_NAME}" is already up to date: {hash}')
```

Comment:
We should not implement this. If this file changes, we should apply it via a migration. More than likely, there is some other db migration that is required before we can re-apply this clone-schema function, so this shouldn't be handled dynamically.

Reply:
Ok, I'll remove it.